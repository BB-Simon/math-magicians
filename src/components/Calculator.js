import Button from './Button';
import charecters from './charecters';

const Calculator = () => (
  <div className="calculator-container">
    <div className="calculator-header">
      <span>0</span>
    </div>
    <div className="item-container">
      {charecters && charecters.map(({ char, name }) => (
        <Button
          key={name}
          className={`item item-${name}`}
          title={char}
        />
      ))}
    </div>
  </div>
);

export default Calculator;
