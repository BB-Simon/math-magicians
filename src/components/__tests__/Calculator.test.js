import TestRenderer from 'react-test-renderer';
import Calculator from '../Calculator';

describe('Jest Snapshot testing suite', () => {
  it('renders correctly', () => {
    const domTree = TestRenderer.create(<Calculator />).toJSON();
    expect(domTree).toMatchSnapshot();
  });
});
