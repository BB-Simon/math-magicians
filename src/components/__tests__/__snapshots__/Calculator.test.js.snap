// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest Snapshot testing suite renders correctly 1`] = `
<div
  className="calculator-container"
>
  <div
    className="calculator-header"
  >
    <span>
      0
    </span>
  </div>
  <div
    className="item-container"
  >
    <button
      className="item item-ac"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="item item-plus-minus"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="item item-remainder"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="item item-divider"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
    <button
      className="item item-seven"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="item item-eight"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="item item-nine"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="item item-multiply"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="item item-four"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="item item-five"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="item item-six"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="item item-substruct"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="item item-one"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="item item-two"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="item item-three"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="item item-add"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="item item-zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="item item-dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="item item-equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
